File: src/main/java/com/sisqueslabs/nexora/api/contexts/chat/application/port/ModelsGateway.java
package com.sisqueslabs.nexora.api.contexts.chat.application.port;

import com.sisqueslabs.nexora.api.contexts.chat.domain.entities.Model;

/**
 * Resolves whether a requested model exists in the catalog. Today it's
 * implemented by infrastructure.mock; tomorrow an HTTP client to
 * nexora-models, without the application layer changing.
 */
public interface ModelsGateway {

    Model resolve(String name);
}

File: src/contexts/chat/application/ports/models.gateway.ts
import { ModelAggregate } from '@contexts/chat/domain/aggregates/model.aggregate';

export const MODELS_GATEWAY = Symbol('MODELS_GATEWAY');

/**
 * Resolves whether a requested model exists in the catalog. Today it's
 * implemented by infrastructure/mock; tomorrow an HTTP client to
 * nexora-models, without the application layer changing.
 */
export interface IModelsGateway {
  resolve(name: string): Promise<ModelAggregate>;
}
